### ISSUE TITLE
[master] crash at disk size checking

### ISSUE THREAD

**Issue by wllacer** [archlinux/archinstall#674]
Using master branch. 
Starting the process (after getting keyboard and country data) archinstall crashes with a ValueError exception.
```Select one of the above regions to download packages from (by number or full name): 57
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/werner/projects/archinstall-master/archinstall/__main__.py", line 4, in <module>
    archinstall.run_as_a_module()

  File "/home/werner/projects/archinstall-master/archinstall/__init__.py", line 103, in run_as_a_module
    script.execute()

  File "/home/werner/projects/archinstall-master/archinstall/lib/profiles.py", line 178, in execute
    self.spec.loader.exec_module(sys.modules[self.namespace])

  File "<frozen importlib._bootstrap_external>", line 850, in exec_module

  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed

  File "/home/werner/projects/archinstall-master/archinstall/examples/guided.py", line 371, in <module>
    ask_user_questions()

  File "/home/werner/projects/archinstall-master/archinstall/examples/guided.py", line 113, in ask_user_questions
    archinstall.arguments['harddrives'] = archinstall.generic_multi_select(archinstall.all_disks(),

  File "/home/werner/projects/archinstall-master/archinstall/lib/user_interaction.py", line 152, in generic_multi_select
    printed_options.append(f'{option}')

  File "/home/werner/projects/archinstall-master/archinstall/lib/disk.py", line 255, in __repr__
    return f"BlockDevice({self.device_or_backfile}, size={self.size}GB, free_space={'+'.join(part[2] for part in self.free_space)}, bus_type={self.bus_type})"

  File "/home/werner/projects/archinstall-master/archinstall/lib/disk.py", line 398, in size
    return self.convert_size_to_gb(device['size'])

  File "/home/werner/projects/archinstall-master/archinstall/lib/disk.py", line 391, in convert_size_to_gb
    return float(units.get(unit, lambda s : None)(size[:-1]))

  File "/home/werner/projects/archinstall-master/archinstall/lib/disk.py", line 385, in <lambda>
    'G' : lambda s : float(s),

ValueError: could not convert string to float: '298,1'
```

**Comment by wllacer** [comment]
I sent a PR moving the size output to bytes, so it might get into the next release. Output gets a bit odd because there are to many decimals sometimes, and some rounding issues (0.9999... vs 1) I imagine it's unavoidable.
I made a small mistake in my original correction and found a series of issues with fully formatted output, So i cut the gordian knot and went to byte size directly.
OTOH the original convert_to_gb function had an even worse error. All calculations outside M G and T where wrong. The original code multiplied / divided by 1024 times the index relative to G, and it should had been 1024 to the power of the index, i.e. 1024**2 for T, 1024**3 for G, etc ...

**Comment by Torxed** [comment]
Yea that wasn't the best pieces of code in my career hehe. I'll merge it in time for release, I will just have to make sure all the sizes and stuff work out correctly when passing it to `parted` mainly. As we use positions in a unit that I can't remember from the top of my head. But I'll test this properly before release while I fix the remaining tasks surrounding partitioning.

I released RC1 yesterday, so testing has begun to weed out any remaining stability issues.

**Comment by wllacer** [comment]
Besides the fact that it shows a ten decimal fraction, in my tests I haven't found any problems with parted even with some weird (inexact) sizes. 
If i've read the logs correctly you define the last partition with 100% as the end size, filling the disk; so the exact size (down to the byte) of a disk is not that relevant. Which is the minimum size of a disk you expect ? If it's over a gigabyte (IIRC the installation guide says a bit under 2 GiB), and both assumptions are correct, i can change the code with no consequences to round to a mere decimal, so output doesn't look too cluttered.

**Comment by Torxed** [comment]
Fixed in #695
